src/engine/image.cpp

```diff
@@ -3106,4 +3106,51 @@ namespace fheroes2
             }
         }
     }
+
+    void ApplyVerticalGradient( fheroes2::Image & image, const uint8_t outsideColor, const uint8_t insideColor, const uint8_t borderWidth, const uint8_t borderColor )
+    {
+        const int32_t height = image.height();
+        const int32_t width = image.width();
+
+        uint8_t * inData = image.image();
+        uint8_t * inTransform = image.transform();
+
+        fheroes2::Image outImage;
+        fheroes2::Copy( image, outImage );
+        uint8_t * outData = outImage.image();
+        uint8_t * outTransform = outImage.transform();
+
+        const uint8_t center_y = static_cast<uint8_t>( std::max( 1, ( height / 2 ) - height % 2 ) );
+        const float dColor = (float)( outsideColor - insideColor );
+
+        for ( uint8_t row = 0; row < height; row++ ) {
+            float heightScale = static_cast<float>( abs( center_y - row ) ) / center_y;
+
+            uint8_t val = static_cast<uint8_t>( abs( insideColor + (int)( heightScale * dColor ) ) );
```

Comment:
We can slightly simplify this code:
```suggestion
        const uint8_t dColor = outsideColor - insideColor;

        for ( uint8_t row = 0; row < height; row++ ) {
            const uint8_t heightScale = ( dColor * static_cast<uint8_t>( std::abs( centerY - row ) ) ) / centerY;

            const uint8_t val = static_cast<uint8_t>( std::abs( insideColor + heightScale ) );
```